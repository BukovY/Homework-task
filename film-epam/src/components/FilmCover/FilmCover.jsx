import React from "react";
import styles from "./FilmCover.module.sass";
import { getFilmCover } from "../../utils/functrions";

const FilmCover = ({ img, rating }) => {
  const ratingToRender = rating;
  return (
    <div className={styles.card}>
      <div
        className={
          rating >= 7
            ? styles.rating_up
            : rating > 1
            ? styles.rating_down
            : styles.rating_hide
        }
      >
        {ratingToRender && ratingToRender.toFixed(1)}
      </div>
      <div className={styles.film_cover}>
        <img
          src={getFilmCover(img)}
          className={styles.film_cover}
          alt="cover"
        />
      </div>
    </div>
  );
};

export default FilmCover;
