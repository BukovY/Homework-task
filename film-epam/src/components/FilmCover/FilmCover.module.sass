@mixin rating
  position: relative
  top: 35px
  left: -5px
  width: 30px
  text-align: center
  border-radius: 5px

.film_cover
  width: 100%
  height: auto
  cursor: pointer

.film_cover img
  width: 100%
  border-radius: 5px

.rating_up
  @include rating
  background-color: green

.rating_down
  @include rating
  background-color: gray

.rating_hide
  @include rating
  color: #00ff0000
  background-color: #00ff0000
