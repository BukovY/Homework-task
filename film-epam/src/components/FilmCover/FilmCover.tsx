import React, { FC } from "react";
import { getFilmCover } from "../../utils/functrions";
import classNames from "classnames/bind";
import { MovieDetailsInterface } from "../../types/movie";
import { makeStyles } from "@material-ui/core/styles";

type MovieDetailsPropsType = {
  el: MovieDetailsInterface;
};

const useStyles = makeStyles(() => ({
  filmCover: {
    width: "100%",
    height: "auto",
    img: {
      width: "100%",
      borderRadius: "5px",
    },
  },
  rating: {
    position: "relative",
    top: "35px",
    left: "-5px",
    width: " 30px",
    textAlign: "center",
    borderRadius: "5px",
    zIndex: 1,
  },
  ratingUp: {
    backgroundColor: "green",
  },
  ratingDown: {
    backgroundColor: "gray",
  },
  ratingHide: {
    color: "#00ff0000",
    backgroundColor: "#00ff0000",
  },
}));

export const FilmCover: FC<MovieDetailsPropsType> = ({ el }) => {
  const classes = useStyles();
  const ratingToRender = el.vote_average;
  const cx = classNames.bind(classes);
  const filmCardClass = cx(
    "rating",
    { ratingUp: el.vote_average >= 7 },
    { ratingDown: el.vote_average > 1 && el.vote_average < 7 },
    { ratingHide: el.vote_average <= 1 }
  );
  return (
    <div>
      <div className={filmCardClass}>
        {ratingToRender && ratingToRender.toFixed(1)}
      </div>
      <div className={classes.filmCover}>
        <img
          src={getFilmCover(el.poster_path)}
          className={classes.filmCover}
          alt="cover"
        />
      </div>
    </div>
  );
};
