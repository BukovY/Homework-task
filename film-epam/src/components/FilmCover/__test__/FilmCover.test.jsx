import React from "react";
import rerender from "react-test-renderer";
import { FilmCover } from "../FilmCover";

const el = {
  title: "Title film",
  vote_average: 7.8,
};
describe("Testing <FilmCover/> Component", () => {
  it("renders correctly", () => {
    const tree = rerender.create(<FilmCover el={el} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
