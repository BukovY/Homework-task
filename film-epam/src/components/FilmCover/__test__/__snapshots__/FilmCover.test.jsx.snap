// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <FilmCover/> Component renders correctly 1`] = `
<div>
  <div
    className="rating rating_up"
  >
    7.8
  </div>
  <div
    className="film_cover"
  >
    <img
      alt="cover"
      className="film_cover"
      src="https://upload.wikimedia.org/wikipedia/commons/6/65/No-Image-Placeholder.svg"
    />
  </div>
</div>
`;
